content/courses/onchain-development/local-setup.md

```diff
@@ -17,59 +18,85 @@ description:
 
 ## Lesson
 
-There's no lesson here! Let's install Solana CLI tools, the Rust SDK, and
-Anchor, and create a test program to ensure that our setup works.
+This lesson is a guide to installing the tools required for developing onchain
+programs. Let's install Solana CLI tools, the Rust SDK, and Anchor, and create a
+test program to ensure that our setup works.
 
 ## Lab
 
-#### Extra steps for Windows users
+### Extra steps for Windows users
 
-Firstly install
-[Windows Terminal](https://apps.microsoft.com/detail/9N0DX20HK701) from the
-Microsoft store.
+> macOS and Linux users can skip this section. Linux and macOS are the
+> recommended operating systems for Solana program development. However, if
+> you're on Windows, you can still follow along with these extra steps.
 
-Then
+Firstly, make sure you have Windows Terminal installed, otherwise you can
+install it from the
+[Microsoft store](https://apps.microsoft.com/detail/9N0DX20HK701).
+
+Then,
 [install Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install).
 WSL provides a Linux environment that launches instantly whenever you need it
 and doesn't slow your computer down.
 
 Start Windows Terminal, launch an 'Ubuntu' session inside the terminal, and
 proceed with the rest of these steps.
 
-#### Download Rust
+### Download Rust
 
 First, download Rust by
 [following the instructions](https://www.rust-lang.org/tools/install):
 
-```
+```bash
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
 ```
 
-#### Download the Solana CLI tools
+### Download the Solana CLI tools
 
-Next
+Next,
 [download the Solana CLI tools](https://docs.solana.com/cli/install-solana-cli-tools).
 
-```
-sh -c "$(curl -sSfL https://release.anza.xyz/stable/install)"
+Solana installation no longer supports symbolic channel names (`edge`, `beta`,
+`stable`), hence we'll have to specify the version.
+
+> the command below will install version `1.18.22` of the Solana CLI.
+
+```bash
+sh -c "$(curl -sSfL https://release.anza.xyz/v1.18.22/install)"
 ```
 
 Afterwards, `solana -V` should show `solana-cli 1.18.x` (any number for `x` is
-fine).
+fine) which is the version of the Solana CLI installed in your system indicating
+that it was installed successfully.
 
-#### Download Anchor
+### Download Anchor
 
-Finally [download Anchor](https://www.anchor-lang.com/docs/installation):
+Finally, [download Anchor](https://www.anchor-lang.com/docs/installation):
 
-```
+```bash
 cargo install --git https://github.com/coral-xyz/anchor avm --locked --force
+```
+
```

Comment:
Do we know what error message people would get if they didn't have these dependencies? Can we add it?

Reply:
I'll spin up a VM to see what errors one may get.

Reply:
I checked in VM. The error was not specific to anchor or solana cli. It was more of linux and rust related. The compilation would fail without those dependencies, as those are important dev dependencies on linux.

Reply:
> The compilation would fail without those dependencies

That's what I suspected - so `anchor build` would throw an error? Write what the error is.

Reply:
Going to resolve this, as someone else already won the bounty for this lesson and these changes are still excellent editions.